SO-93

hashemJaber opened the pull request:
- Added Page for adding receipts provided by a  USER for a SPECIFIC PROJECT

- Added Page with OCR technogoligy that automates information extracting via Taggun API

- Added Page with editable table for information edit in case of OCR failure

hashemJaber committed:
SO-93
-Added Reciept item creation form
-Added Reciept item extractor
-Added Reciept item list form

hashemJaber committed:
SO-93
- Page for adding receipts provided by a  USER for a SPECIFIC PROJECT

-Page with OCR technogoligy that automates information extracting via Taggun API

-Page with editable table for information edit in case of OCR failure

jvniorrr commented on src/app/(dashboard)/invoices/page.tsx:
```diff
@@ -0,0 +1,421 @@
+			method: 'POST',
+			headers: {
+				accept: 'application/json',
+				apikey: 'f0d78cd0be8511eeb72409b0b60cbbed',
```
not suer if this should be public?!?!

jvniorrr replied:
![UHM](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExdGxyN3preDBtY3dudGdpaG5nOW16M2M4enlocWVibWFkNWtpajc2NCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/DhiRqIsofVMi7fWNBQ/giphy.gif)

hashemJaber replied:
yeah I am aware the issue was that when ever I used to pull from process.env it would cause an error until I realized that its a client component, any tips for fixing ?

hashemJaber replied:
![](https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExaG8xOHhycGpwamYzd3QzNmdzZHZ1enR4MHhveWJkemU2Mmg5NWt4eSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/TdWQV0lWuRVyR1JxuY/giphy.gif)

jvniorrr commented on src/app/(dashboard)/invoices/page.tsx:
```diff
@@ -0,0 +1,421 @@
+				},
+			]);
+
+		entryError ? alert(JSON.stringify(entryError)) : console.log('');
```
no alerts... :)

hashemJaber replied:
thank for catching this for me

jvniorrr commented on src/app/(dashboard)/invoices/itemsList.tsx:
```diff
@@ -0,0 +1,153 @@
+	};
+
+	return (
+		<div style={{ maxHeight: '400px', overflowY: 'auto' }}>
```
Assume this is when the receipt is interpretted... Thought we was omitting that part, as far as trying to retrieve item per item deal? Should likely ask the team on this.

hashemJaber replied:
Yes we did omit this but the work was already done and for now the code is not in use so it shouldn't cause any issues hopefully

jvniorrr commented on src/app/(dashboard)/invoices/page.tsx:
```diff
@@ -0,0 +1,421 @@
+	const [imageURL, setImageURL] = useState(DEFAULT_IMAGE);
+	//FIXME: EXTRACT PROJECT_ID AMD org_id dyncmically
+
+	const [reciept, setReciept] = useState<IReceipts | null | any>({
```
should it not be empty? uhm

jvniorrr replied:
the object

jvniorrr commented on src/app/(dashboard)/invoices/page.tsx:
```diff
@@ -0,0 +1,421 @@
+
+				itemsInfo.forEach((item: any) => {
+					try {
+						console.log(
```
console.log statement

jvniorrr commented on src/app/(dashboard)/invoices/page.tsx:
```diff
@@ -0,0 +1,421 @@
+						});
+						setItems(tmps);
+					} catch (e: any) {
+						alert('failed due to\n:' + e);
```
remove alerts

jvniorrr commented on src/app/(dashboard)/invoices/page.tsx:
```diff
@@ -0,0 +1,421 @@
+
+				setReceiptData(response);
+				setLoading(false);
+				alert(JSON.stringify(response));
```
remove alerts

hashemJaber replied:
yes sir 


![](https://media1.giphy.com/media/v1.Y2lkPTc5MGI3NjExOTd6dHl6ZWF3dmltYWRuOGNtNzM0OXQ4NzN4M2I5dWhpNjh0YjZkayZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/j1zfofiWuk7rZgQH6B/giphy.gif)

jvniorrr commented on src/app/(dashboard)/invoices/page.tsx:
```diff
@@ -0,0 +1,421 @@
+
+			if (error) {
+				console.error(error);
+				alert(error);
```
remove alerts

hashemJaber replied:
yes my pharoh 

![](https://media1.giphy.com/media/v1.Y2lkPTc5MGI3NjExZjVvYWlkZG1pdjJmN2w2b2o5ZGtqc2JhcnI5OXlrbG5leHA2M3huNiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/l1JoiCvR7XVddZrPO/giphy.gif)

jvniorrr commented on src/app/(dashboard)/invoices/page.tsx:
```diff
@@ -0,0 +1,421 @@
+				},
+			]);
+
+		entryError ? alert(JSON.stringify(entryError)) : console.log('');
```
remove alerts

hashemJaber replied:
yes my lord


![](https://media1.giphy.com/media/v1.Y2lkPTc5MGI3NjExOGlxY3FjZXFkbGFyYWhlbW41bjRsMmQ5emxueDlsZW1vNGZrM2hydSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/JERMKtKC9uJrzgvtoz/giphy.gif)

jvniorrr commented on src/app/(dashboard)/invoices/page.tsx:
```diff
@@ -0,0 +1,421 @@
+			? console.error('error creating receipt ' + entryError)
+			: console.info('Creating reciept worked');
+
+		if (
```
span text popup likely if their is any error at all. generalized error messaging indicating to retry or refresh.

Next sprint on this though.

jvniorrr commented on src/app/(dashboard)/invoices/page.tsx:
Will the CSS as is, but please follow our code styling on our codebase. We have been using separate CSS filing convention since last semester 😃 so lets try avoid using inline styling please. However code functioning as expected seems so.

hashemJaber replied:
![](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExZ3NiYWV4emNzNGc4dXptZWJlZjFzYTlyNTZhMDB3OTMzdjJ1eHR2OCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/KnkA0AoLMcaxGQg3nb/giphy.gif)

hashemJaber committed:
SO-93
Added Page for adding receipts provided by a USER for a SPECIFIC PROJECT

Added Page with OCR technogoligy that automates information extracting via Taggun API

Added Page with editable table for information edit in case of OCR failure

hashemJaber committed:
SO-93
Added Page for adding receipts provided by a USER for a SPECIFIC PROJECT

Added Page with OCR technogoligy that automates information extracting via Taggun API

Added Page with editable table for information edit in case of OCR failure
